Add Deployment Engineering to our handbook (#1226)
mattermost/mattermost-handbook: operations/deployment-engineering/frequently-asked-questions.md

# Deployment Engineering: Frequently asked questions

## Why is this team important now?

Retention is the #1 priority, and underdeployment is the leading cause for contraction. 

> Deployment Engineering helps TAMs by leveraging engineering talent to accelerate customer success with deeper platform knowledge given their background in software design engineering, thus accelerating seat deploymets, thus reducing contraction, thus increasing renewal rates.

Reviewer: A question, not a comment can we add a term for all customer facing teams (instead of helps TAMs) since I could see in the workflow chart that the team will be aligning with all customer facing teams

Author: @adityadav1987 Good feedback! Adjusted to

> Deployment Engineering helps customer-facing teams (TAMs, CREs, CS) by...

Let me know if this works on your end

Reviewer: looks good

> # Deployment Engineering: Frequently Asked Questions

Reviewer 2: ```suggestion
# Deployment Engineering: Frequently asked questions
```

## How is this team measured?

Primary measurement is **Customer Seats Deployed** - % of customers with >$50K ARR with at least 80% of seats deployed.

## How will work be allocated to this team?

> Deployment Engineering will work closely with other post-sales teams (support, customer success, TAMs) to determine the highest impact technical customer priorities for the team to work on.

Reviewer 3: ```suggestion
Deployment Engineering will work closely with other post-sales teams (CREs, customer success, TAMs) to determine the highest impact technical customer priorities for the team to work on.
```

Most solutions should not require long-term maintenance, as they will be standalone scripts, plugins or other code that is not dependent on Mattermost server versions.

For solutions that require maintenance, we will discuss ownership between Deployment Engineering, TAMs, R&D and a vendor on a case-by-case basis.

## How does this role differ from TAMs (Technical Account Managers)?

TAMs are responsible for ongoing relationship management with technical champions, and own pre-defined customer onboarding in standard deployment environments (ie. follow our gold standard deployments).

On the other hand, Deployment Engineering supports non-standard deployments with complex technical problems, as well as work closely with customers on temporary one-off custom solutions & integrations based on [requirements outlined by the CSMs/TAMs](https://docs.google.com/document/d/1EbP4Ab7N2hEAv--DRCcZHk0XtlG1MFP8F8F0viaTM7I/edit).

Thus, Deployment Engineering acts as an extension of R&D supporting post-sales and customer success on the most complex, technical problems, with expertise on how to run and deploy Mattermost and the platform as a whole.

> Finally, we encourage a representative from TAM to join weekly Deployment Engineering meetings, and vice versa, to help deepen collaboration & understanding of each team’s workstreams.

Reviewer 3: should this be on the handbook? maybe it can be rephrased into what's expected instead of a call to action.

Author: Thanks for the feedback! Removing for now

## Why does this role not report to Customer Success, like TAMs?

TAMs are responsible for ongoing relationship management, and are evaluated against renewal rates.

Deployment Engineering engages with customers on a short-term basis, in a technical capacity. Their evaluation is not based on renewal rates, and hence do not directly report to the revenue organization (Sales & Customer Success).

> ## How does this role differ from Support Engineers & Customer Reliability Engineers?

Reviewer 3: ```suggestion
## How does this role differ from Customer Reliability Engineers?
```

> Support & Customer Reliability Engineers provide first-line of support & troubleshooting, and may fix bugs & security issues, while Deployment Engineering is a point of contact for critical deployment escalations only for customers over $50k ARR (e.g. performance degradation).

Reviewer 3: ```suggestion
Customer Reliability Engineers provide first-line of support & troubleshooting, and may fix bugs & security issues, while Deployment Engineering is a point of contact for critical deployment escalations only for customers over $50k ARR (e.g. performance degradation).
```

## Why does this role not report to Customer Support, if they work on escalations?

Support Engineers are evaluated against CSAT scores and SLA coverage. Deployment Engineering is evaluated against Customer Seats Deployed.

> Moreover, Deployment Engineering is not a support-function, but instead their primary ownership is to help customers run and deploy Mattermost at scale.

Reviewer 3: > help customers run and deploy Mattermost at scale.

this might be hard to differentiate from CRE's objectives, it probably needs more details.

Author: I've removed the second sentence as it's likely not needed here. The differing goals helps differentiate the two functions.

> ## Will this team be a call or SET rotation?

Reviewer 2: ```suggestion
## Will this team be on call or part of SET rotation?
```

> No.

Reviewer 3: Aren't large upgrades happening on weekends? I understand that there is not going to be a regular on-call, but I'd expect DE to be on call during those upgrade windows

Author: True. Though I consider that working days on the weekend more-so than a formal on-call rotation

Reviewer: > True. Though I consider that working days on the weekend more-so than a formal on-call rotation

Jason, probably this might be a longer run thing and really something that should be open to discussion, but I do believe there should be some alert/pager duty(not necessarily on-call) or some other solution to alert a CS/DE engineer if things really go south/ we encounter a sev1 outage kind of issue during odd hours (non-US/EU timings or on weekend) and the issue goes beyond the CRE scope during the triaging.                                                Case in point was a recent case with Bofa (which surfaced during apac hours and the one caused by DR job bringing down their MM instance) where I could see no alerting mechanism for other customer facing teams. We might lose out some valuable response time in case we have to reach out to other customer facing teams (thankfully we were to able to triage and do an initial investigation on that case). Again, this is just a headsup thing and doesn't have to be implemented immediately.

> ## Why is this team important now?

Reviewer 3: I think this section should go along with how it is measured so it stresses the importance and meaning of the metric.

Author: Thanks! Moved it together with the 'how is this team measured' section

Deployment Engineering helps TAMs by leveraging engineering talent to accelerate customer success with deeper platform knowledge given their background in software design engineering, thus accelerating seat deploymets, thus reducing contraction, thus increasing renewal rates.

## What is out of scope for Deployment Engineering?

The following is out of scope for Deployment Engineering

 - Regular customer support (tier 1, 2, 3, 4 support) owned by Customer Support as per our standard support terms: https://mattermost.com/support-terms/ 
 - Customer account management including expansion, renewals, or license management, owned by CSMs and TAMs.
 - Predefined onboarding journey including sales hand-off, standard deployments with standard data migrations, testing, training and end-user rollout, owned by TAMs unless specifically escalated to Deployment Engineering.
 - Identification of key integration opportunities for the Mattermost team to develop, owned by CSMs and TAMs
 - Development or maintenance of in-product features, integrations or other such product functionality, owned by R&D Engineering

>    - Note: In-product feature development may be completed by Deployment Engineering at an exception, e.g. building a custom solution that we decide to integrate into the product. Such a development requires prior approval from Product Engineering, to ensure they can support the feature once developed for the customer.

Reviewer 2: ```suggestion
   - Note: In-product feature development may be completed by Deployment Engineering as an exception, e.g. building a custom solution that we decide to integrate into the product. Such a development requires prior approval from Product Engineering, to ensure they can support the feature once developed for the customer.
```
